### ISSUE TITLE
[Bug]: Bedrock Converse Chat with Tools is throwing error in latest release 0.4.2

### ISSUE THREAD

**Issue by sharat910** [run-llama/llama_index#17317]
### Bug Description

We've been using the Bedrock Converse API and function calling (tool support). The recent release of `llama-index-llms-bedrock-converse` (0.4.2) seems to have broken the interaction with bedrock. It works in 0.4.1. 

Link to recent closed PR related to the release: https://github.com/run-llama/llama_index/pull/17281

P.S. This is my first bug report. Please let me know if I've missed out on anything.

### Version

0.12.6

### Steps to Reproduce

You can use the following simple example (AWS account and sso setup required) to test it out. `llama-index-llms-bedrock-converse` 0.4.1 correctly returns the assistant message with tool calls but 0.4.2 errors out with the trace shared below. The llama-index-core version was pinned to 0.12.6 (latest).

```
from llama_index.llms.bedrock_converse import BedrockConverse
from llama_index.core.tools import FunctionTool
from llama_index.core.prompts import ChatMessage, MessageRole
from llama_index.core.memory import ChatMemoryBuffer

llm = BedrockConverse(
    model="us.anthropic.claude-3-5-haiku-20241022-v1:0",
    profile_name="dev",
    region_name="us-east-1",
)

def get_weather(city: str):
    if city == "Seattle":
        return "20 C, rainy"
    elif city == "Sydney":
        return "22 C, sunny"
    else:
        return "Unknown"


tools = [FunctionTool.from_defaults(get_weather)]

mem = ChatMemoryBuffer.from_defaults(llm=llm)

mem.put(
    ChatMessage(
        role=MessageRole.SYSTEM,
        content="You are a helpful weather assistant.",
    )
)
mem.put(ChatMessage(role=MessageRole.USER, content="What is the weather in Sydney?"))

resp = llm.chat_with_tools(
    tools,
    chat_history=mem.get(),
)
```

### Relevant Logs/Tracbacks

```shell
Traceback (most recent call last):
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/temp.py", line 34, in <module>
    resp = llm.chat_with_tools(
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/llama_index/core/llms/function_calling.py", line 47, in chat_with_tools
    response = self.chat(**chat_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/llama_index/core/instrumentation/dispatcher.py", line 321, in wrapper
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/llama_index/core/llms/callbacks.py", line 173, in wrapped_llm_chat
    f_return_val = f(_self, messages, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/llama_index/llms/bedrock_converse/base.py", line 310, in chat
    response = converse_with_retry(
               ^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/llama_index/llms/bedrock_converse/utils.py", line 355, in converse_with_retry
    return _conversion_with_retry(**converse_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/tenacity/__init__.py", line 336, in wrapped_f
    return copy(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/tenacity/__init__.py", line 475, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/tenacity/__init__.py", line 398, in <lambda>
    self._add_action_func(lambda rs: rs.outcome.result())
                                     ^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/.local/share/uv/python/cpython-3.11.9-macos-aarch64-none/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/.local/share/uv/python/cpython-3.11.9-macos-aarch64-none/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/tenacity/__init__.py", line 478, in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/llama_index/llms/bedrock_converse/utils.py", line 353, in _conversion_with_retry
    return client.converse(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/botocore/client.py", line 569, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/botocore/client.py", line 980, in _make_api_call
    request_dict = self._convert_to_request_dict(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/botocore/client.py", line 1047, in _convert_to_request_dict
    request_dict = self._serializer.serialize_to_request(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sharat/projects/Rebel-Alliance/trash-compactor/llama-index-workflow/.venv/lib/python3.11/site-packages/botocore/validate.py", line 381, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Unknown parameter in input: "tools", must be one of: modelId, messages, system, inferenceConfig, toolConfig, guardrailConfig, additionalModelRequestFields, additionalModelResponseFieldPaths
```

**Comment by logan-markewich** [comment]
@sharat910 it seems like this traceback comes from a notebook right? It seems to be cut off.

There should be an option in your notebook to view the output as a scrollable element or something like that. Having the full traceback would simplify debugging a lot (it could be an issue with the boto3 library, the api for converse updated, or an issue with llama-index code, I can't tell just yet)

**Comment by logan-markewich** [comment]
@schniber you might be able to help as well (I don't have access to easily test bedrock lol)

**Comment by sharat910** [comment]
> @sharat910 it seems like this traceback comes from a notebook right? It seems to be cut off.
> 
> There should be an option in your notebook to view the output as a scrollable element or something like that. Having the full traceback would simplify debugging a lot (it could be an issue with the boto3 library, the api for converse updated, or an issue with llama-index code, I can't tell just yet)

Ah sorry about that! I've updated the trace by running that in a regular python script.

**Comment by schniber** [comment]
Hey all

@sharat910: Thanks a lot for spotting this !

I just filed a Pull Request to fix this regression (#17364)

issue was that the following condition to exclude tool use while merging the kwargs was wrong:

```python
converse_kwargs = join_two_dicts(
        converse_kwargs,
        {
            k: v
            for k, v in kwargs.items()
            if (
                k != "tools"
                or k != "guardrail_identifier"
                or k != "guardrail_version"
                or k != "trace"
            )
        },
    )
```

so an easy fix was to rewrite the exclusion condition like this:

```python
converse_kwargs = join_two_dicts(
        converse_kwargs,
        {
            k: v
            for k, v in kwargs.items()
            if k not in ["tools", "guardrail_identifier", "guardrail_version", "trace"]
        },
    )
```

sorry for the delay in taking care of the regression I had introduced.

Wishing you a happy holidays season !

Thanks ! 

Bests

**Comment by sharat910** [comment]
Thanks @schniber for addressing the issue! Happy Holidays 🎆 !
